core-java-modules/core-java-concurrency-collections-2/src/main/java/com/baeldung/concurrent/lock/ConcurrentAccessBenchmark.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.concurrent.lock;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Thread)
+@Fork(value = 1)
+@Warmup(iterations = 0)
+public class ConcurrentAccessBenchmark {
+    ConcurrentAccessExperiment accessMyMap;
+    static final int SLOTS = 4;
+    static final int THREADS = 10000;
+    static final int BUCKETS = Runtime.getRuntime().availableProcessors() * SLOTS;
+
+    @Param({"Single Lock", "Striped Lock"})
+    private String lockType;
+
+    @Param({"HashMap", "ConcurrentHashMap"})
+    private String mapType;
+
+    @Setup
+    public void setup() {
+        switch (lockType) {
+            case "Single Lock":
+                accessMyMap = new SingleLock();
+                break;
+            case "Striped Lock":
+                accessMyMap = new StripedLock(BUCKETS);
+                break;
+        }
+    }
+
+    @Benchmark
```

Comment:
I believe that having more than one benchmark method would simplify both `ConcurrentAccessBenchmark` and `ConcurrentAccessExperiment`. For example:

```java
@Benchmark
public Map<String, String> singleLockHashMap() {
    return this.single.doWork(new HashMap<>(), THREADS, SLOTS);
}

@Benchmark
public Map<String, String> stripedLockHashMap() {
    return this.striped.doWork(new HashMap<>(), THREADS, SLOTS);
}

@Benchmark
public Map<String, String> singleLockConcurrentHashMap() {
    return this.single.doWork(new ConcurrentHashMap<>(), THREADS, SLOTS);
}

@Benchmark
public Map<String, String> stripedLockHashMap() {
    return this.striped.doWork(new ConcurrentHashMap<>(), THREADS, SLOTS);
}
```

Then, you wouldn't need the `setup` method in this class nor in the experiment class. You could also then remove the `lockType` parameter and wouldn't need a direct reference to `ConcurrentAccessExperiment`.

Reply:
Done!